src/eureka/S3_data_reduction/plots_s3.py

```diff
@@ -565,3 +568,307 @@ def median_frame(data, meta):
+    plt.ylim(0, flux.shape[0])
+    plt.xlabel('x pixels')
+    plt.ylabel('y pixels')
+
+    plt.legend()
+    plt.tight_layout()
+    file_number = str(m).zfill(int(np.floor(np.log10(meta.num_data_files))+1))
+    int_number = str(i).zfill(int(np.floor(np.log10(meta.n_int))+1))
+
+    fname = (f'figs{os.sep}fig3306_file{file_number}_int{int_number}_2D_Frame'
+             + figure_filetype)
+    plt.savefig(meta.outputdir + fname, dpi=250, tight_layout=True)
+    if not meta.hide_plots:
+        plt.pause(0.2)
+
+
+def phot_2d_frame_zoom(meta, m, i, data):
+    """
+    Plots the 2D frame together with the centroid position and apertures.
+    """
+    plt.figure(3504, figsize=(6, 5))
+    plt.clf()
+    plt.suptitle('2D frame with centroid and apertures (zoom-in)')
+    flux, centroid_x, centroid_y = data.flux[i], data.centroid_x[i], data.centroid_y[i]
+
+    xmin, xmax = data.flux.x.min().values-meta.xwindow[0], data.flux.x.max().values-meta.xwindow[0]
+    ymin, ymax = data.flux.y.min().values-meta.ywindow[0], data.flux.y.max().values-meta.ywindow[0]
+
+    im = plt.imshow(flux, vmin=0, vmax=5e3, origin='lower', aspect='equal',
+                    extent=[xmin, xmax, ymin, ymax])
+    plt.scatter(centroid_x, centroid_y, marker='x', s=25, c='r', label='centroid')
+    plt.title('Full Frame')
+    plt.ylabel('y pixels')
+    plt.xlabel('x pixels')
+
+    circle1 = plt.Circle((centroid_x, centroid_y), meta.photap, color='r',
+                         fill=False, lw=3, alpha=0.7, label='target aperture')
+    circle2 = plt.Circle((centroid_x, centroid_y), meta.skyin, color='w',
+                         fill=False, lw=4, alpha=0.8, label='sky aperture')
+    circle3 = plt.Circle((centroid_x, centroid_y), meta.skyout, color='w',
+                         fill=False, lw=4, alpha=0.8)
+    plt.gca().add_patch(circle1)
+    plt.gca().add_patch(circle2)
+    plt.gca().add_patch(circle3)
+
+    add_colorbar(im, label='Flux (electrons)')
+    xlim_min = max(0, centroid_x - meta.skyout - 10)
+    xlim_max = min(centroid_x + meta.skyout + 10, flux.shape[1])
+    ylim_min = max(0, centroid_y - meta.skyout - 10)
+    ylim_max = min(centroid_y + meta.skyout + 10, flux.shape[0])
+
+    plt.xlim(xlim_min, xlim_max)
+    plt.ylim(ylim_min, ylim_max)
+    plt.xlabel('x pixels')
+    plt.ylabel('y pixels')
+
+    plt.legend()
+    plt.tight_layout()
+    file_number = str(m).zfill(int(np.floor(np.log10(meta.num_data_files))+1))
+    int_number = str(i).zfill(int(np.floor(np.log10(meta.n_int))+1))
+
+    fname = (f'figs{os.sep}fig3504_file{file_number}_int{int_number}_2D_Frame_Zoom'
+             + figure_filetype)
+    plt.savefig(meta.outputdir + fname, dpi=250, tight_layout=True)
+    if not meta.hide_plots:
+        plt.pause(0.2)
+
+
+def phot_2d_frame_oof(meta, m, i, data, flux_w_oof):
+    """
+    Plots the 2D frame together with the centroid position and apertures.
+    """
+    fig, ax = plt.subplots(2, 1, figsize=(8.2, 4.2))
```

Comment:
You'll also want to set the figure number to 3307 here as well so that the figure gets overridden with each new copy. And then also add a clf after that to make sure the figure gets cleared out before each plot